### PR TITLE
[email-async] Messages are sent asynchronously

### PR DESCRIPTION
**Pull request by hallelujah**
Part of THREESCALE-7574 #comment we split it into multiple pull-request so it is easier to review

### CODE REVIEW

**Review by akostadinov — DISMISSED** [review]
Sorry, review was in pending mode, forgot to submit it.

File: test/unit/message_test.rb [hunk 1]
```diff
@@ -119,409 +119,409 @@ def test_send_notifications
...
+    buyer = FactoryBot.create(:buyer_account, provider_account: provider)
+    provider_sent_message = FactoryBot.create(:message, sender: provider)
+    buyer_received_message = FactoryBot.create(:received_message, message: provider_sent_message, receiver: buyer)
+    buyer_sent_message = FactoryBot.create(:message, sender: buyer)
+    provider_received_message = FactoryBot.create(:received_message, message: buyer_sent_message, receiver: provider)
+
+    [provider_sent_message, provider_received_message].each(&:hide!)
+
+    provider_sent_message.restore_for!(provider)
+    refute provider_sent_message.reload.hidden?
```
Comment (josemigallas) [inline test/unit/message_test.rb:522]:
```suggestion
    assert_not provider_sent_message.reload.hidden?
```

File: test/unit/message_test.rb [hunk 2]
```diff
@@ -119,409 +119,409 @@ def test_send_notifications
...
+
+    buyer_sent_message.restore_for!(provider)
+    refute provider_received_message.reload.hidden?
```
Comment (josemigallas) [inline test/unit/message_test.rb:525]:
```suggestion
    assert_not provider_received_message.reload.hidden?
```

**Review by josemigallas — COMMENTED** [review]
Fixed outdated usages for you if you'd like to merge them.

**Review by akostadinov — APPROVED** [review]

File: test/unit/logic/plan_changes_test.rb [hunk 3]
```diff
@@ -91,23 +101,27 @@ def setup
...
-    msg = @app.buyer_changes_plan!(@new_paid_plan)
+    perform_enqueued_jobs(only: ActionMailer::DeliveryJob) do
+      msg = @app.buyer_changes_plan!(@new_paid_plan)
 
-    assert_equal "Please enter your credit card before changing the plan.", msg
-    assert_equal 'Old plan', @app.plan.name
-    assert_empty ActionMailer::Base.deliveries
+      assert_equal "Please enter your credit card before changing the plan.", msg
+      assert_equal 'Old plan', @app.plan.name
+      assert_empty ActionMailer::Base.deliveries
```
[the remark this answers is not in the document]
Comment (hallelujah) [reply test/unit/logic/plan_changes_test.rb:109]: Well it does not matter really. Just cosmetic

**Pull request merged**